onadata/libs/utils/csv_builder.py

```diff
@@ -533,17 +533,21 @@ def _build_ordered_columns(cls, survey_element, ordered_columns,
                 # generated when we reindex
                 ordered_columns[child.get_abbreviated_xpath()] = None
 
-    def _update_columns_from_data(self, cursor):
+    def _update_ordered_columns_from_data(self, cursor):
+        """
+        Populates `self.ordered_columns` object that is
+        used to generate export column headers for
+        forms that split select multiple and gps data.
+        """
         # add ordered columns for select multiples
         if self.split_select_multiples:
-            for key, choices in self.select_multiples.items():
+            for (key, choices) in iteritems(self.select_multiples):
                 # HACK to ensure choices are NOT duplicated
-                if key in self.ordered_columns.keys():
-                    self.ordered_columns[key] = \
-                        remove_dups_from_list_maintain_order(
-                            [choice.replace('/' + name, '/' + label)
-                             if self.show_choice_labels else choice
-                             for choice, name, label in choices])
+                self.ordered_columns[key] = \
+                    remove_dups_from_list_maintain_order(
+                        [choice.replace('/' + name, '/' + label)
+                         if self.show_choice_labels else choice
+                         for choice, name, label in choices])
```

Comment:
The if check was removed, is there a change here that gives us confidence hat we will not encounter a `KeyError` exception or are we handling the exception elsewhere?

Reply:
Ah, that was unintentionally removed. I've reinstated that check.
Thanks for catching that.